mdm-core/grails-app/controllers/uk/ac/ox/softeng/maurodatamapper/core/file/ThemeImageFileController.groovy

```
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2020-2023 University of Oxford and Health and Social Care Information Centre, also known as NHS Digital
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package uk.ac.ox.softeng.maurodatamapper.core.file
+
+import uk.ac.ox.softeng.maurodatamapper.core.admin.ApiProperty
+import uk.ac.ox.softeng.maurodatamapper.core.admin.ApiPropertyService
+import uk.ac.ox.softeng.maurodatamapper.core.controller.EditLoggingController
+import uk.ac.ox.softeng.maurodatamapper.security.User
+import uk.ac.ox.softeng.maurodatamapper.util.Utils
+
+import grails.gorm.transactions.Transactional
+import groovy.util.logging.Slf4j
+
+
+@Slf4j
+class ThemeImageFileController extends EditLoggingController<ThemeImageFile> {
+
+    static responseFormats = ['json', 'xml']
+
+    ThemeImageFileService themeImageFileService
+    ApiPropertyService apiPropertyService
+
+    ThemeImageFileController() {
+        super(ThemeImageFile)
+    }
+
+    @Override
+    def show() {
+        ThemeImageFile resource = null
+        if (params.apiPropertyId && themeImageFileService.apiPropertyHasImage(Utils.toUuid(params.apiPropertyId))) {
+            resource = themeImageFileService.findByApiPropertyId(params.apiPropertyId)
```

I think that ideally if the API property itself doesn't exist, that should return an appropriate error message, I think you should be able to use a call to notFound(ApiProperty, params.apiPropertyId) to return in that case.

---

I can see this was updated, but I think the `ApiProperty` arg should be included to get the correct class name in the error message.

---

ApiProperty added